src/Raven.Server/Web/Studio/DataDirectoryInfo.cs

```diff
@@ -90,6 +96,38 @@ public async Task UpdateDirectoryResult(string databaseName)
             }
         }
 
+        public static bool CanAccessPath(string folderPath, out string error)
+        {
+            var originalFolderPath = folderPath;
+            while (true)
+            {
+                var directoryInfo = new DirectoryInfo(folderPath);
+                if (directoryInfo.Exists == false)
```

Comment:
Can this throw?

Reply:
No. It will return false if any error occurs while trying to determine if the path exists.